lib/redmine/menu_manager/top_menu_helper.rb

```diff
@@ -173,13 +173,26 @@ def render_module_top_menu_node(items = more_top_menu_items)
                               title: I18n.t("label_modules"),
                               test_selector: "op-app-header--modules-menu-button",
                               "aria-label": I18n.t("label_modules"))
-        items.each do |item|
-          menu.with_item(tag: :a,
-                         href: url_for(item.url),
-                         label: item.caption,
-                         test_selector: "op-menu--item-action") do |menu_item|
-            menu_item.with_leading_visual_icon(icon: item.icon) if item.icon
-          end
+
+        item_groups.each do |item_group|
+          render_menu_item_group(menu, item_group)
+        end
+      end
+    end
+  end
+
+  def render_menu_item_group(menu, item_group)
+    menu.with_group do |menu_group|
+      menu_group.with_heading(title: item_group[:title], align_items: :flex_start) if item_group[:title]
+
+      item_group[:items].each do |item|
+        menu_group.with_item(
+          tag: :a,
```

Comment:
There is no hover effect on the list items which is a strong indication that something went wrong. Looking at the rendered DOM, there is a strange composition of `a` tags instead of `li` with an `a` inside.
<img width="300" alt="Bildschirmfoto 2024-07-18 um 08 35 05" src="https://github.com/user-attachments/assets/00e6d468-1e78-44f2-b829-4a24ad2113b7">

That happens because the group open a new ActionList internally. While looking at the Primer repo, I found [this place](https://github.com/primer/view_components/blob/main/app/components/primer/alpha/action_menu/list.rb#L93), in which the tag is passed through to the content, but not removed from the global system_arguments (which I consider a bug, but who knows). However, I also found [this place](https://github.com/primer/view_components/blob/main/app/components/primer/alpha/action_list/item.rb#L247), which indicates that an `href` is enough to transform it into an anchor. So removing the tag attribute here, should fix the issue.

Reply:
Works wonderful. Thanks for the catch.